src/components/dbCalls.js

```diff
@@ -82,19 +83,46 @@ const updateLogsForRoom = async (newLog, roomID) => {
     }
 }
 
-//fetches a player's targets from database
+//fetches a player's targets from database including openseason
 const fetchTargetsForPlayer = async (playerName, roomID) => {
     try {
         const playerCollectionRef = collection(db, 'rooms', roomID, 'players');
         const playerQuery = query(playerCollectionRef, where('name', '==', playerName));
-        const playerSnapshot = await getDocs(playerQuery);
-        const playerTargets = playerSnapshot.docs[0].data().targets;
-        return playerTargets;
-    }    
-    catch (error) {
+        const openSeasonQuery = query(playerCollectionRef, where('openSeason', '==', true));
+
+        const [playerSnapshot, openSeasonSnapshot] = await Promise.all([
+            getDocs(playerQuery),
+            getDocs(openSeasonQuery),
+        ]);
+
+        let targets = [];
+        if (!playerSnapshot.empty) {
+            const playerTargets = playerSnapshot.docs[0].data().targets || [];
+            const openSeasonPlayers = openSeasonSnapshot.docs.map(doc => doc.data().name);
+            let foundIndex = -1;
+
+            for (let k = 0; k < openSeasonPlayers.length; k++) { // For each open season player
+                if (playerName === openSeasonPlayers[k]) { // If same name
+                    foundIndex = k;
+                    break; // Exit the loop once found
+                }
+            }
+
+            if (foundIndex !== -1) {
+                const newOpenSzn = openSeasonPlayers.slice(0, foundIndex).concat(openSeasonPlayers.slice(foundIndex + 1));
```

Comment:
You can try array splice to make this even more concise, it's a fun function: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toSpliced